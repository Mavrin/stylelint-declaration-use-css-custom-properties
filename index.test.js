const path = require("path");
const stylelint = require("stylelint");

function resolveFile(name) {
  return path.resolve(__dirname, `./css-examples/${name}`);
}

function extractErrorMessage(warnings) {
  return warnings.map(({ text }) => text);
}

describe("stylelint plugins", () => {
  it("should pass for valid css file", () => {
    return stylelint
      .lint({
        files: [resolveFile("valid.pcss")],
        configFile: resolveFile("config.js"),
      })
      .then(({ results }) => {
        expect(results).toHaveLength(1);
        expect(results[0].warnings).toHaveLength(0);
        expect(extractErrorMessage(results[0].warnings)).toMatchSnapshot();
      });
  });
  it("should throw error for invalid css file", () => {
    return stylelint
      .lint({
        files: [resolveFile("invalid.pcss")],
        configFile: resolveFile("config.js"),
      })
      .then(({ results }) => {
        expect(results).toHaveLength(1);
        expect(results[0].warnings).toHaveLength(4);
        expect(extractErrorMessage(results[0].warnings)).toMatchSnapshot();
      });
  });
});
