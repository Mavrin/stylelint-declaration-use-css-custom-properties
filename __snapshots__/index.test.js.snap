// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelint plugins should pass for valid css file 1`] = `Array []`;

exports[`stylelint plugins should throw error for invalid css file 1`] = `
Array [
  "Please use css custom property for next css definition \`color\` in \\"color: black\\" (mavrin/stylelint-declaration-use-css-custom-properties)",
  "Please use css custom property for next css definition \`color\` in \\"border: 1px solid #fff\\" (mavrin/stylelint-declaration-use-css-custom-properties)",
  "Please use css custom property for next css definition \`length\` in \\"border: 1px solid #fff\\" (mavrin/stylelint-declaration-use-css-custom-properties)",
  "Please use css custom property for next css definition \`color\` in \\"background-color: rgba(37, 156, 154, 0.4)\\" (mavrin/stylelint-declaration-use-css-custom-properties)",
]
`;
